### ISSUE TITLE
"padded-blocks" rule: add the option "consistent"

### ISSUE THREAD

**Issue by OliverSieweke** [eslint/eslint#11702]
**What rule do you want to change?**

[`padded-blocks`
](https://eslint.org/docs/rules/padded-blocks)

**Does this change cause the rule to produce more or fewer warnings?**

Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**

Adding a new possible value `"consistent"` (in addition to `"always"` and `"never"`).

**Please provide some example code that this change will affect:**

```js
\\ No Padding:
if (a) {
    b();
}

\\ Padding:
if (a) {

    b();

}

\\ Inconsistent Padding:
if (a) {
    b();

}
```

**What does the rule currently do for this code?**

With `["error", "always"]`:

```js
\\ No Padding: Bad!
if (a) {
    b();
}

\\ Padding: Good!
if (a) {

    b();

}

\\ Inconsistent Padding: Bad!
if (a) {
    b();

}
```

With `["error", "never"]`:

```js
\\ No Padding: Good!
if (a) {
    b();
}

\\ Padding: Bad!
if (a) {

    b();

}

\\ Inconsistent Padding: Bad!
if (a) {
    b();

}
```

**What will the rule do after it's changed?**

With `["error", "consistent"]`:

```js
\\ No Padding: Good!
if (a) {
    b();
}

\\ Padding: Good!
if (a) {

    b();

}

\\ Inconsistent Padding: Bad!
if (a) {
    b();

}
```

**Are you willing to submit a pull request to implement this change?**

Yes

**Comments**

This rule change would be in line with similar `"consistent"` options that already exist for rules such as [`array-bracket-newline`](https://eslint.org/docs/rules/array-bracket-newline), [`object-curly-newline`](https://eslint.org/docs/rules/object-curly-newline) or [`function-paren-newline`](https://eslint.org/docs/rules/function-paren-newline).

**Comment by kaicataldo** [comment]
This seems reasonable to me. Just to clarify, what would the behavior be when `allowSingleLineBlocks: true`?

**Comment by OliverSieweke** [comment]
At the moment single-line-blocks are allowed if:

- no-padding is enforced (value `"never"`), regardless of the option `allowSingleLineBlocks`.
- padding is enforced (value `"always"`) and the option `allowSingleLineBlocks` is set to `true`. 

Given that `"consistent"` would be more permissive than both `"never"` and `"always"`, I think it would make most sense to allow single-line-blocks regardless of the `allowSingleLineBlocks` option, as is currently also the case for `"never"`. 
`allowSingleLineBlocks` would then remain an option that would only affect the rule if it is set to `"always"`.

**Comment by ilyavolodin** [comment]
Does `consistant` option apply across the blocks within a file, or on a single block only? It's unclear based on the examples provided.

**Comment by OliverSieweke** [comment]
`consistent` would not apply across the blocks within a file. At least that's how I see the option being the most useful. That's also how things currently work with the `consistent` option of [`array-bracket-newline`](https://eslint.org/docs/rules/array-bracket-newline) and [`object-curly-newline`](https://eslint.org/docs/rules/object-curly-newline) for example.

So with `["error", "consistent"]` the following would be fine _within the same file_:
``` js
\\ Consistent with padding - all good!
if (a) {

    b();

}

\\ Consistent without padding - all good!
if (a) {
    b();
}
```

**Comment by ilyavolodin** [comment]
OK, thanks. I guess it makes sense considering the other rules.
